website/docs/server/spring-server/spring-properties.md

```diff
@@ -31,4 +31,4 @@ details on the supported configuration properties.
 | graphql.subscriptions.keepAliveInterval     | **Deprecated**. Keep the websocket alive and send a message to the client every interval in ms. Defaults to not sending messages | null                          |
 | graphql.subscriptions.protocol              | WebSocket based subscription protocol. Supported protocols: APOLLO_SUBSCRIPTIONS_WS / GRAPHQL_WS                                 | GRAPHQL_WS                    |
 | graphql.batching.enabled                    | Boolean flag indicating whether to enable custom dataloader instrumentations for 1 or more GraphQL Operations                    | false                         |
-| graphql.batching.strategy                   | Configure which custom dataloader instrumentation will be used (LEVEL_DISPATCHED or SYNC_EXHAUSTION)                             | LEVEL_DISPATCHED              |
+| graphql.batching.strategy                   | Configure which custom dataloader instrumentation will be used                                                                   | SYNC_EXHAUSTION               |
```

Comment:
since we only have single data loader batching strategy do we really need this config?

Reply:
i would like to keep the door open for future dispatch mechanisms, `SYNC_EXHAUSTION` works but comes with a price, needs to hold state of each field, and memory usage could be a concern, in the future we could try to have a different dispatch mechanism that will use less memory maybe.

Reply:
The keyword is "might" -> Unless there is something already in the works/roadmap I am unsure if there is ever going to be one so I'd just drop it and reintroduce it later on if needed.

Reply:
Also I find `graphql.batching.enabled` property confusing -> we always support batching so this only applies to dataloader config (as per comments). Wondering whether this should be deprecated in favor of something like `graphql.loader.batching.strategy = SYNC_EXHAUSTION | NONE` (`NONE` would imply current `graphql.batching.enabled = false`)

Reply:
>we always support batching
we added `graphql.batching.enabled` in order to opt out from built-in "by level" graphql-java dataloader instrumentation.

https://github.com/ExpediaGroup/graphql-kotlin/blob/master/servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/server/spring/GraphQLSchemaConfiguration.kt#L90

Reply:
yeah we added it originally for the custom data loaders -> just wondering whether we should revisit this property naming for the next major